### DISCUSSION TITLE
Role Restricted Workflows

### DISCUSSION THREAD

**Discussion by TLMondo** [directus/directus#20296] [Feature Requests]
### Summary

In combination to the request for more event triggers (https://github.com/directus/directus/discussions/20295) for the creation of Compliance/Governance workflows, some of those workflows should have extra restrictions on them. A flag of "Protected" could be applied, and only users in the `Protected` role could modify.

In some system there is a declared `Super Admin` that can only be modified by the Customer Success Manager for the client, and that user is the only one that can make certain changes (e.g. modify users in the `Protected` role).

### Basic Example

See above

### Motivation

Protection of Compliance/Governance monitoring/enforcing workflows

### Detailed Design

See above

### Requirements List

See above

### Drawbacks

None?

### Alternatives

Risk sensitive enforcement to be modified without protection.

### Adoption Strategy

Should be safe to roll out to all.

For the eventual "My Super admin is gone" issues, have a "General Support Alias" email required on signup. Have a "reset admin" request form, that then sends instructions to that alias. Issues beyond that need a CSM (Customer Success Manager) intervention.

For self-hosted, make it part of the configuration files, it's on them to keep that up-to-date.

**Comment by rijkvanzanten** [comment]
I'm a little confused as to what we're talking about here.. 

> for the creation of Compliance/Governance workflows, some of those workflows should have extra restrictions on them.

Do you mean flows?

**Reply by TLMondo** [reply]
Meaning if I create a workflow that enforces/monitors PII restrictions, then it is a very sensitive/mission-critical workflow, and not just anybody should be able to touch it.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
